src/js/views/maps/LegendView.js

```diff
@@ -361,218 +331,182 @@ define([
+        // is displayed in the tooltip on mouseover. See the
+        // rect.on('mousemove'... function, below
+        data = data.sort((a, b) => a.value - b.value);
+        const min = data[0].value;
+        const max = data[data.length - 1].value;
+        const range = max - min;
+        const roundingConstant = Utilities.getRoundingConstant(range);
+
+        // SVG element
+        const svg = this.createSVG({
+          dropshadowFilter: false,
+          width,
+          height,
+        });
+
+        // Add the preview class and dropshadow to the SVG
+        svg.classed(this.classes.previewSVG, true);
+        svg.style("filter", "url(#dropshadow)");
+
+        // Create a gradient using the data
+        const gradient = svg
+          .append("defs")
+          .append("linearGradient")
+          .attr("id", gradientId)
+          .attr("x1", "0%")
+          .attr("y1", "0%");
+
+        const getOffset = (d) => `${((d.value - min) / range) * 100}%`;
+        const getStopColor = (d) => {
+          const r = d.color.red * 255;
+          const g = d.color.green * 255;
+          const b = d.color.blue * 255;
+          return `rgb(${r},${g},${b})`;
+        };
+
+        // Add the gradient stops
+        data.forEach((d) => {
+          gradient
+            .append("stop")
+            // offset should be relative to the value in the data
+            .attr("offset", getOffset(d, data))
+            .attr("stop-color", getStopColor(d));
+        });
+
+        // Create the rectangle
+        const rect = svg
+          .append("rect")
+          .attr("x", 0)
+          .attr("y", 0)
+          .attr("width", width)
+          .attr("height", gradientHeight)
+          .attr("rx", gradientHeight * 0.1)
+          .style("fill", `url(#${gradientId})`);
+
+        // Create a proxy element to attach the tooltip to, so that we can move the
+        // tooltip to follow the mouse (by moving the proxy element to follow the mouse)
+        const proxyEl = svg.append("rect").attr("y", gradientHeight);
+
+        rect
+          .on("mousemove", () => {
+            if (view.model.get("visible")) {
+              // Get the coordinates of the mouse relative to the rectangle
+              let xMouse = d3.mouse(this)[0];
+              if (xMouse < 0) {
+                xMouse = 0;
+              }
+              if (xMouse > width) {
+                xMouse = width;
+              }
+              // Get the relative position of the mouse to the gradient
+              const relativePosition = xMouse / width;
+              // Get the value at the relative position by interpolating the data
+              let value = d3.interpolate(
+                data[0].value,
+                data[data.length - 1].value,
+              )(relativePosition);
+              // Show tooltip with the value
+              if (value || value === 0) {
+                // Round or show in scientific notation
+                if (roundingConstant) {
```

Comment:
This duplicates logic from the helper, can we use the Utilities function instead?

Reply:
Same as above- the goal for LegendView here is to maintain a noop with minimum effort